### DISCUSSION TITLE
WPF incompatibility (INPC006)

### DISCUSSION THREAD

**Discussion by OZ1** [canton7/PropertyChanged.SourceGenerator#33] [Q&A] [answered]
**Description**
I get INPC006 when used inside Window derived class:
warning INPC006: Found one or more methods called 'OnPropertyChanged' to raise the PropertyChanged event, but they had an unrecognised signatures or were inaccessible. No PropertyChanged events will be raised from this type
warning CS0067: Событие "MainWindow.PropertyChanged" никогда не используется.

**To Reproduce**
Create WPF project.
Add any `[Notify]` field to MainWindow.xaml.cs.
Compile.

**Version Info**
 - PropertyChanged.SourceGenerator version: 1.0.6
 - Roslyn Version: "4.4.0-3.22518.13 (7856a68c)". Версия языка: 10.0.

**Additional Info**
class `Window` is derived from `DependencyObject` which contains `protected virtual void OnPropertyChanged(DependencyPropertyChangedEventArgs e)`

**Comment by canton7** [comment, marked as answer]
It's not usual to implement INPC in your views. You've got DependencyProperties available there. If you need to do that, add your own OnPropertyChanged method which has a compatible signature.

**Reply by canton7** [reply]
Although, thinking about it, it would make sense to only complain about unknown OnPropertyChanged methods when the type implements INotifyPropertyChanged. If it doesn't implement INPC, the OnPropertyChanged method surely has to be unrelated...

**Reply by canton7** [reply]
Fixed in 5eb3c21c45be267248528e5a9fbff8ec93caf197
